### ISSUE TITLE
Support code highlighting in package documentation

### ISSUE THREAD

**Issue by tmattio** [ocaml/ocaml.org#99]
A potential approach would be to parse the HTML we receive from the doc pipeline with `Lambdasoup` and inject the highlighted code from `Hilite`. If we're satisfied with the approach, we may consider upstreaming this to `odoc` directly once `hilite` is stable. cc @jonludlam

**Comment by patricoferris** [comment]
It might be worth looking at the `hilite` package that we use for the tutorials @jonludlam to see if a similar approach could be taken in odoc when generating the HTML the site consumes :)) ?

**Comment by tmattio** [comment]
I was actually thinking about that - it'd remove the need to include highlight.js in odoc!

**Comment by patricoferris** [comment]
Yeah that would be good, and wherever README.md.html is generated (I couldn't track that down... somewhere in docs-ci or voodoo maybe) applying `Hilite.Md.transform` would bring the highlighting to the READMEs too :))

**Comment by tmattio** [comment]
@jonludlam Should we open an issue on odoc to discuss this?

**Comment by Angelvicks** [comment]
Hello @tmattio , I am an Outreachy Applicant and I will like to work on this issue? How can I get started please?

**Comment by tmattio** [comment]
Hi @Angelvicks. Given the discussion above, I don't think this is a good first issue, so I'll remove the tag. Sorry for the confusion! I'll let you pick another issue marked as "good-first-issue". Could you also confirm that you are up and running and can run the website locally? We are asking applicants to set their environment up before taking an issue.

**Comment by Angelvicks** [comment]
Alright @tmattio I will start by running the website locally then.

**Comment by emillon** [comment]
Hi!
I just pushed some [docs for dune libraries](https://ocaml.org/p/dune-configurator/latest/doc/index.html) and I noticed that there's no highlighting in the examples. Is this the same issue?
The source is at https://github.com/ocaml/dune/blob/355319919b18023bfee818db1f8a42f5e9bc2c68/otherlibs/configurator/src/index.mld#L25 - should I add `@ocaml` or is this unsupported?
Thanks

**Comment by favonia** [comment]
Hi, I want to follow up on @emillon comment---I hope `@ocaml` is not be needed for `{[ ... ]}` (if that's the case). And, @tmattio, unless I miss something, it seems embedding a JavaScript can solve all the parsing problems, and this approach has been adopted by `odoc`. Here are the three steps to adopt it:

1. Download a copy of `highlight.pack.js` from https://highlightjs.org/download with OCaml and common languages, and then embed it somewhere. (Note that `odoc` now supports language tags, so we can expect code blocks other than OCaml.) For example, `odoc` by default puts a packed script with at least OCaml under `_doc/_html/odoc.support/`.
2. Add the following two lines to the HTML template:
   ```html
   <script src="/path/to/highlight.pack.js"></script>
   <script>hljs.highlightAll();</script>
   ```
   `odoc` is already generating the needed CSS class information to make the highlighting script work. For example, the code block pointed out by @emillon already has the `language-ocaml` CSS class. The highlighting script will do its job.
   
   _For impatient people:_ one can run the following code in the browser console to inject a minimum version that supports OCaml and get a preview:
   ```javascript
   function loadScript(src) {
     var script = document.createElement("script");
     script.src = src;
     document.body.appendChild(script);
   }
   loadScript("https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.9.0/highlight.min.js");
   loadScript("https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.9.0/languages/ocaml.min.js");
   hljs.highlightAll();
   ```
   One can see that the highlighting code is working. The result will not look great because the CSS is not customizing the colors for highlight.js to match the overall theme, which is the next item---
3. Follow the [official documentation of highlight.js](https://highlightjs.readthedocs.io/en/latest/theme-guide.html) to choose a consistent and beautiful set of colors.

I wonder if I missed anything? If everything above is correct, I do feel this is a good first issue. @tmattio If you find this plan agreeable, I would also suggest updating your top comment so that newcomers who want to work on this will not be distracted by `Lambdasoup` or other in-house highlighting solutions.
